src/Paprika/Merkle/Node.cs

```diff
@@ -0,0 +1,81 @@
+using System.Runtime.InteropServices;
+using Paprika.Crypto;
+using Paprika.Data;
+
+namespace Paprika.Merkle;
+
+public enum NodeType : byte
+{
+    Leaf,
+    Extension,
+    Branch
+}
+
+[StructLayout(LayoutKind.Explicit, Pack = 1, Size = Size)]
+public readonly struct Branch
```

Comment:
A Branch is 35 bytes long if we always want to keep the Keccak (which we initially do):
- 1 byte for `Header`, which contains `Dirty` and `NodeType`
- 2 bytes for `Nibbles`, a Bit Set of nibbles
- 32 bytes for `Keccak` (initially, always present)